Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/FileTransporter.cs

```diff
@@ -97,44 +98,27 @@ public FileTransferResult Run(CancellationToken cancellationToken)
+
                     // Check the fingerprint of the server if given.
-                    if (!String.IsNullOrEmpty(_batchContext.Connection.ServerFingerPrint))
+                    if (!string.IsNullOrEmpty(expectedServerFingerprint))
                     {
+                        SetCurrentState(TransferState.Connection, "Checking server fingerprint.");
                         try
                         {
-                            SetCurrentState(TransferState.Connection, "Checking server fingerprint.");
-                            // If this check fails then SSH.NET will throw an SshConnectionException - with a message of "Key exchange negotiation failed".
-                            client.HostKeyReceived += delegate (object sender, HostKeyEventArgs e)
-                            {
-                                // First try with SHA256 typed fingerprint
-                                using (SHA256 mySHA256 = SHA256.Create())
-                                {
-                                    var sha256Fingerprint = Convert.ToBase64String(mySHA256.ComputeHash(e.HostKey));
-                                    // Set the userResultMessage in case checking server fingerprint failes.
-                                    userResultMessage = $"Can't trust SFTP server. The server fingerprint does not match. " +
-                                                            $"Expected fingerprint: '{_batchContext.Connection.ServerFingerPrint}', but was: '{sha256Fingerprint}'";
-                                    e.CanTrust = (sha256Fingerprint == _batchContext.Connection.ServerFingerPrint);
-                                }
-
-                                if (!e.CanTrust)
-                                {
-
-                                    userResultMessage = $"Can't trust SFTP server. The server fingerprint does not match. " +
-                                                            $"Expected fingerprint: '{_batchContext.Connection.ServerFingerPrint}', but was: '{BitConverter.ToString(e.FingerPrint).Replace("-", ":")}'";
-                                    // If previous failed try with MD5 typed fingerprint
-                                    var expectedFingerprint = Util.ConvertFingerprintToByteArray(_batchContext.Connection.ServerFingerPrint);
-                                    e.CanTrust = e.FingerPrint.SequenceEqual(expectedFingerprint);
-                                }
-                                    
-                            };
+                            CheckServerFingerprint(client, expectedServerFingerprint);
                         }
                         catch (Exception e)
                         {
-                            _logger.NotifyError(null, "Error when checking the server fingerprint: ", e);
+                            _logger.NotifyError(null, $"Error when checking the server fingerprint", e);
```

Comment:
you can del the $ in string :)

Reply:
Also, down below e.g. on `SftpPathNotFoundException` you include the `ex.Message` in the error text, but here you do not - is this intended like this?

Reply:
Might've been just left out by mistake. Added the ex.Message to the logger.NotifyError